### PR TITLE
feat(GaussDB): fix gaussdb opengauss instance

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  fix gaussdb opengauss instance
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  fix gaussdb opengauss instance
```

## PR Checklist

* [ ] Tests added/passed.

```
make testacc TEST=./huaweicloud/services/acceptance/gaussdb/ TESTARGS='-run TestAccOpenGaussInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run TestAccOpenGaussInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_basic
=== PAUSE TestAccOpenGaussInstance_basic
=== CONT  TestAccOpenGaussInstance_basic
--- PASS: TestAccOpenGaussInstance_basic (5747.86s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   5747.904s


make testacc TEST=./huaweicloud/services/acceptance/gaussdb/ TESTARGS='-run TestAccOpenGaussInstance_haModeCentralized'  
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run TestAccOpenGaussInstance_haModeCentralized -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_haModeCentralized
=== PAUSE TestAccOpenGaussInstance_haModeCentralized
=== CONT  TestAccOpenGaussInstance_haModeCentralized
--- PASS: TestAccOpenGaussInstance_haModeCentralized (2251.53s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   2251.574s



make testacc TEST=./huaweicloud/services/acceptance/gaussdb/ TESTARGS='-run TestAccOpenGaussDatabase_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run TestAccOpenGaussDatabase_basic -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussDatabase_basic
=== PAUSE TestAccOpenGaussDatabase_basic
=== CONT  TestAccOpenGaussDatabase_basic
--- PASS: TestAccOpenGaussDatabase_basic (2127.22s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   2127.263s
```

* [ ] Documentation updated.
* [ ] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
删了后查询
    ab. Related resources (parent resources) not found
    \>>>>>> Paste the screenshot here <<<<<<
    -->
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
删除后再删除.
    bb. Related resources (parent resources) not found
    \>>>>>> Paste the screenshot here <<<<<<
    -->

### CODE REVIEW

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go [hunk 1]
```diff
@@ -513,378 +427,718 @@ func resourceOpenGaussInstanceCreate(ctx context.Context, d *schema.ResourceData
...
 		endpoints = append(endpoints, endpoint)
 	}
 
-	return multierror.Append(nil,
-		d.Set("private_ips", ipList),
-		d.Set("endpoints", endpoints),
-	).ErrorOrNil()
+	var errs []error
+	errs = append(errs, d.Set("private_ips", ipList))
+	errs = append(errs, d.Set("endpoints", ipList))
```
Comment (saf3dfsa) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go]:
```suggestion
	errs = append(errs, d.Set("endpoints", endpoints))
```

File: docs/resources/gaussdb_opengauss_instance.md [hunk 2]
```diff
@@ -278,5 +301,25 @@ This resource provides the following timeouts configuration options:
 OpenGaussDB instance can be imported using the `id`, e.g.
 
 ```bash
-$ terraform import huaweicloud_gaussdb_opengauss_instance.test 1f2c4f48adea4ae684c8edd8818fa349in14
+$ terraform import huaweicloud_gaussdb_opengauss_instance.test <id>
+```
+
+Note that the imported state may not be identical to your resource definition, due to the attribute missing from the
+API response. The missing attribute is: `password`, `ha.0.mode`, `ha.0.instance_mode`, `configuration_id`,
```
Comment (Jason-Zhang9309) [inline docs/resources/gaussdb_opengauss_instance.md]:
```suggestion
API response. The missing attributes include: `password`, `ha.0.mode`, `ha.0.instance_mode`, `configuration_id`,
```

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go [hunk 3]
```diff
@@ -513,378 +427,718 @@ func resourceOpenGaussInstanceCreate(ctx context.Context, d *schema.ResourceData
...
+		d.Set("charging_mode", utils.PathSearch("charge_info.charge_mode", instance, nil)),
+		d.Set("ha", flattenGaussDBOpenGaussResponseBodyHa(instance)),
+		d.Set("datastore", flattenGaussDBOpenGaussResponseBodyDatastore(instance)),
+		d.Set("backup_strategy", flattenGaussDBOpenGaussResponseBodyBackupStrategy(instance)),
+	)
+
+	var dnNum = 1
+	mErr = multierror.Append(mErr, setOpenGaussNodesAndRelatedNumbers(d, instance, &dnNum)...)
+	mErr = multierror.Append(mErr, d.Set("volume", flattenGaussDBOpenGaussResponseBodyVolume(instance, dnNum)))
+	mErr = multierror.Append(mErr, setOpenGaussPrivateIpsAndEndpoints(d, instance)...)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go]: put this part at line 605

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go [hunk 4]
```diff
@@ -513,378 +427,718 @@ func resourceOpenGaussInstanceCreate(ctx context.Context, d *schema.ResourceData
...
+
+func flattenGaussDBOpenGaussResponseBodyHa(instance interface{}) []interface{} {
+	rst := make([]interface{}, 0, 1)
+	rst = append(rst, map[string]interface{}{
+		"mode":             utils.PathSearch("type", instance, nil),
+		"replication_mode": utils.PathSearch("ha.replication_mode", instance, nil),
+		"consistency":      utils.PathSearch("ha.consistency", instance, nil),
+		"instance_mode":    utils.PathSearch("instance_mode", instance, nil),
+	})
+	return rst
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go]:
```suggestion
	rst :=[]interface{}{
	    map[string]interface{}{
		"mode":             utils.PathSearch("type", instance, nil),
		"replication_mode": utils.PathSearch("ha.replication_mode", instance, nil),
		"consistency":      utils.PathSearch("ha.consistency", instance, nil),
		"instance_mode":    utils.PathSearch("instance_mode", instance, nil),
	}
    }
	return rst
```

**Pull request merged**